wpengine/faustjs
# Merl 708 use auth hook

## Tasks


## Description

Introduce the `useAuth`, `useLogin`, and `useLogout` hooks for authentication within a Faust app. Docs on their respective APIs can be found below.



## Related Issue(s):



## Testing

1. Check out the branch
2. Create `pages/auth.js` and add the following content:
    ```jsx
    import { gql, useQuery } from '@apollo/client';
    import { getApolloAuthClient, useAuth } from '@faustwp/core';
    import { useLogout } from '@faustwp/core/dist/mjs';
    
    function AuthenticatedView() {
      const client = getApolloAuthClient();
      const { logout } = useLogout();
    
      const { data, loading } = useQuery(
        gql`
          {
            viewer {
              posts {
                nodes {
                  id
                  title
                }
              }
              name
            }
          }
        `,
        { client },
      );
    
      if (loading) {
        return <>Loading...</>;
      }
    
      return (
        <>
          <p>Welcome {data?.viewer?.name}!</p>
    
          <p>My posts</p>
    
          <ul>
            {data?.viewer?.posts?.nodes.map((post) => (
              <li key={post.id}>{post.title}</li>
            ))}
          </ul>
    
          <button onClick={() => logout()}>Logout</button>
        </>
      );
    }
    
    export default function Page(props) {
      const { isAuthenticated, isReady, loginUrl } = useAuth({
        strategy: 'redirect',
        shouldRedirect: false,
      });
    
      if (!isReady) {
        return <>Loading...</>;
      }
    
      if (isAuthenticated === true) {
        return <AuthenticatedView />;
      }
    
      return (
        <>
          <p>Welcome!</p>
          <a href={loginUrl}>Login</a>
        </>
      );
    }
    ```
3. Ensure you can login, view your authenticated content, and logout



## Screenshots



## Documentation Changes
* [useAuth hook](https://h8wgkzpfgzenqgth1ld3x38a5.js.wpenginepowered.com/docs/reference/useAuth)
* [useLogin hook](https://h8wgkzpfgzenqgth1ld3x38a5.js.wpenginepowered.com/docs/reference/useLogin)
* [useLogout hook](https://h8wgkzpfgzenqgth1ld3x38a5.js.wpenginepowered.com/docs/reference/useLogout)



## Dependant PRs
